protocol/communities/manager.go

```diff
@@ -1545,36 +1545,35 @@ func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, des
 		}
 
 		// A new community, we need to check if we need to validate async.
-		// That would be the case if it has a contract. We queue everything and process separately
-
+		// That would be the case if it has a contract. We queue everything and process separately.
 		if shouldQueue {
 			return nil, m.Queue(signer, community, description.Clock, payload)
 		}
-
 	} else {
-		// we only queue if the clock is greater
-		if shouldQueue && community.config.CommunityDescription.Clock < description.Clock {
+		// only queue if already known control node is different than the signer
+		// and if the clock is greater
+		shouldQueue = shouldQueue && !common.IsPubKeyEqual(community.ControlNode(), signer) && community.config.CommunityDescription.Clock < description.Clock
```

Comment:
Shouldn't this condition be part of `shouldQueue`'s original assignment?

Reply:
Not really, the `community` may be nil there and I din't want to repeat `community != nil` check there.